File: Samples/CslaLight/vb/Rolodex/Business/Server/BusinessClasses/CompanyContactPhoneList.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Csla
Imports Csla.Security
Imports Csla.Core
Imports Csla.Serialization
Imports Csla.Silverlight
Imports Csla.Validation


#If(Not SILVERLIGHT) Then
Imports System.Data.SqlClient
Imports Rolodex.Business.Data
Imports Csla.Data
#End If

Namespace Rolodex.Business.BusinessClasses
  <Serializable> _
  Public Class CompanyContactPhoneList
	  Inherits BusinessListBase(Of CompanyContactPhoneList,CompanyContactPhone)

#If SILVERLIGHT Then
	Public Sub New()
		MarkAsChild()
	End Sub
	Protected Overrides Sub AddNewCore()
	  Dim newContactPhone As CompanyContactPhone = CompanyContactPhone.NewCompanyContactPhone()
	  Add(newContactPhone)
	End Sub

#Else
	Private Sub New()
		MarkAsChild()
	End Sub
	Protected Overrides Function AddNewCore() As Object
	  Dim newContactPhone As CompanyContactPhone = CompanyContactPhone.NewCompanyContactPhone()
	  Add(newContactPhone)
	  Return newContactPhone
	End Function
#End If

	Friend Shared Function NewCompanyContactPhoneList() As CompanyContactPhoneList
	  Return New CompanyContactPhoneList()
	End Function
  End Class
End Namespace

File: Samples/CslaLight/cs/Rolodex/Business/Server/BusinessClasses/CompanyContactPhoneList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Csla;
using Csla.Security;
using Csla.Core;
using Csla.Serialization;
using Csla.Silverlight;
using Csla.Validation;


#if!SILVERLIGHT
using System.Data.SqlClient;
using Rolodex.Business.Data;
using Csla.Data;
#endif

namespace Rolodex.Business.BusinessClasses
{
  [Serializable]
  public class CompanyContactPhoneList : BusinessListBase<CompanyContactPhoneList,CompanyContactPhone>
  {

#if SILVERLIGHT
    public CompanyContactPhoneList() { MarkAsChild(); }
    protected override void AddNewCore()
    {
      CompanyContactPhone newContactPhone = CompanyContactPhone.NewCompanyContactPhone();
      Add(newContactPhone);
    }

#else
    private CompanyContactPhoneList() { MarkAsChild(); }
    protected override object AddNewCore()
    {
      CompanyContactPhone newContactPhone = CompanyContactPhone.NewCompanyContactPhone();
      Add(newContactPhone);
      return newContactPhone;
    }
#endif

    internal static CompanyContactPhoneList NewCompanyContactPhoneList()
    {
      return new CompanyContactPhoneList();
    }
  }
}
